### PR TITLE
Use const& over plain & in Python export wrappers

### PR DESCRIPTION
**Pull request by martyngigg**
**Blocked by: #32465**

**Description of work.**

The Python layer contained many functions cppcheck flagged as being able to accept a `const&` rather than plain `&`. This fixes them for more protection from the compiler. See the commit comment for the `const *const` usage for singletons.

**To test:**

* Code review and all tests should pass.
* Check the cppcheck results have no `constParameterCallback` errors left.

*There is no associated issue.*

*This does not require release notes* because **it is an internal maintenance task.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
Please note that this has been rebased on top of #32496 so needs to wait until that is merged before merging it.

**Review by thomashampson — APPROVED** [review]
Looks good and removes a big chunk of cppcheck warnings.

**Review by DavidFair — APPROVED** [review]

**Pull request merged**
